pkg/api/validation/validation.go

```diff
@@ -38,8 +38,7 @@ func validateVolumes(volumes []api.Volume) (util.StringSet, errs.ValidationError
 	allErrs := errs.ValidationErrorList{}
 
 	allNames := util.StringSet{}
-	for i := range volumes {
-		vol := &volumes[i] // so we can set default values
+	for i, vol := range volumes {
```

Comment:
This is a good way to prove that we are not depending on mutating changes, but it is pessimal for performance - it will make a copy  of every object.  I think we should stick with pointers.